### PR TITLE
Expand error handling

### PR DESCRIPTION
**Pull request by ureeves**
It is important to handle errors properly, especially when these errors occur due to possibly malicious entities - be it contract or callers - trying to attack a long lived instance of the virtual machine.

This PR does this by handling errors appropriately, in particular in the WASM imports.

### CODE REVIEW

File: piecrust/src/session.rs [hunk 1]
```diff
@@ -554,11 +555,11 @@ impl Session {
 
         let mut serializer =
             StandardBufSerializer::new(ser, scratch, Infallible);
-        serializer.serialize_value(value).expect("Infallible");
+        serializer.serialize_value(value)?;
 
         let pos = serializer.pos();
 
-        buf[..pos].to_vec()
+        Ok(buf[..pos].to_vec())
```
Comment (herr-seppia) [inline piecrust/src/session.rs:562]: Ditto

**Review by herr-seppia — APPROVED** [review]
LGTM

Not sure if there is the need of bounds check

File: piecrust/src/imports.rs [hunk 2]
```diff
@@ -174,28 +176,24 @@ fn hd(mut fenv: FunctionEnvMut<Env>, name_ofs: i32, name_len: u32) -> u32 {
...
-                buf[..data.len()].copy_from_slice(&data);
-            });
+    let data = env.meta(&name).unwrap_or_default();
 
-            data.len() as u32
-        }
-        _ => 0u32,
-    }
+    instance.with_arg_buffer(|buf| {
+        buf[..data.len()].copy_from_slice(&data);
```
[the remark this answers is not in the document]
Comment (ureeves) [reply piecrust/src/imports.rs:185]: For this one this is unnecessary. I pushed a commit to make this a little clearer. Metadata items are always `<=` the arg buffer length.

File: piecrust/src/session.rs (hunk 1 above)
Comment (ureeves) [reply piecrust/src/session.rs:562]: If the type is too large to fit in `MAX_META_SIZE` then this will error out anyway. Fortunately

**Pull request merged**